common/src/lib.rs

```diff
@@ -27,4 +29,77 @@ impl BlockHeader {
     pub fn hash(&self) -> Hash256 {
         unimplemented!()
     }
+
+    /// Verifies whether the given block header is a valid successor of this block.
+    ///
+    /// Note that you still need to verify the block body and the finalization proof.
+    pub fn verify_next_block(&self, header: &BlockHeader) -> Result<(), String> {
+        if header.height != self.height + 1 {
+            return Err(format!(
+                "Invalid height: expected {}, got {}",
+                self.height + 1,
+                header.height
+            ));
+        }
+        if header.previous_hash != self.hash() {
+            return Err(format!(
+                "Invalid previous hash: expected {}, got {}",
+                self.hash(),
+                header.previous_hash
+            ));
+        }
+        if !self
```

Comment:
Comparison with the current leader should be done here, right?
Then would you add a TODO comment for that?